runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2994,36 +3001,40 @@ J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator
    TR::Register *temp1Reg = cg->allocateRegister();
    generateRIEInstruction(cg, TR::InstOpCode::AGHIK, node, temp1Reg, targetReg, zeroArraySize);
    generateRXInstruction(cg, TR::InstOpCode::CLG, node, temp1Reg, generateS390MemoryReference(vmThreadReg, offsetof(J9VMThread, heapTop), cg));
-   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BH, node, oolJumpLabel);
+   cursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BH, node, oolJumpLabel);
+   iComment("Branch to oolJumpLabel if there isn't enough space for a 0 size array.");
 
    // If there's enough space, then we can continue to allocate.
    generateRXInstruction(cg, TR::InstOpCode::STG, node, temp1Reg, generateS390MemoryReference(vmThreadReg, offsetof(J9VMThread, heapAlloc), cg));
    
    bool use64BitClasses = comp->target().is64Bit() && !TR::Compiler->om.generateCompressedObjectHeaders();
 
-   // Init class and fields, then jump to end of ICF
+   // Init class field, then jump to end of ICF
    generateRXInstruction(cg, use64BitClasses ? TR::InstOpCode::STG : TR::InstOpCode::ST, node, classReg, generateS390MemoryReference(targetReg, TR::Compiler->om.offsetOfObjectVftField(), cg));
-   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
+   cursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
+   iComment("Init class field and jump to end of ICF.");
 
    // We end up in this region of the ICF if the first dimension is non-zero and the second dimension is zero.
-   generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, nonZeroFirstDimLabel);
+   cursor = generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, nonZeroFirstDimLabel);
+   iComment("nonZeroFirstDimLabel, 2nd dim length is 0.");
 
    TR::Register *componentClassReg = cg->allocateRegister();
    generateRXInstruction(cg, TR::InstOpCode::LG, node, componentClassReg, generateS390MemoryReference(classReg, offsetof(J9ArrayClass, componentType), cg));
 
-   // Calculate maximum allowable object size in elements and jump to OOL if it's higher than firstDimLenReg.
+   // Calculate maximum allowable object size in elements and jump to OOL if it's lower than firstDimLenReg.
```

Comment:
It branches on `COND_BHR` which is "Branch High Register". This comment edit looks incorrect.

Reply:
We compare `firstDimLenReg` to `maxObjectSizeInElements`, branch if `firstDimLenReg` is higher than `maxObjectSizeInElements`. The comment originally said that we would jump if `maxObjectSizeInElements` was higher than `firstDimLenReg`, which wouldn't make sense. Unless I read it wrong?

Reply:
I think the comment was a bit confusing so I have reworded it.

Reply:
Look good now!